etna/core/mixins.py

```diff
@@ -89,6 +90,36 @@ def to_dict(self):
         params["_target_"] = BaseMixin._get_target_from_class(self)
         return params
 
+    def set_params(self, **params: dict) -> "BaseMixin":
+        """Return new object instance with modified parameters.
+
+        The method works on simple estimators as well as on nested objects
+        (such as :class:`~etna.pipeline.Pipeline`). The latter have
+        parameters of the form ``<component>.<parameter>`` so that it's
+        possible to update each component of a nested object.
+
+        Parameters
+        ----------
+        **params: dict
+            Estimator parameters.
+
+        """
+        params_dict = self.to_dict()
+
+        # we need to change specification into list of nested params, "model.depth" to nested structure
```

Comment:
This description is not very easy to understand.
Let's write something like:
```
We need to change specification into list of nested params, e.g. {"model.depth": value} to {"model": {"depth": value}}.
```
I think that we should extract this logic into private method for better readability. For example, `_convert_dict_flat_to_nested`.

Reply:
May be we should also write couple of tests for this conversion.

Reply:
I agree, I'll change the specification and write tests. It is generally a good idea to isolate code performing interpretable side tasks into distinct functions.
The only correction I'd like to make is change from proposed 
> `We need to change specification into list of nested params, e.g. {"model.depth": value} to {"model": {"depth": value}}.`
to 
`We need to change specification into dict of nested params, e.g. {"model.depth": value} to {"model": {"depth": value}}.`
as nested params are indeed stored in a dict, not in a list.

Reply:
I'm going to create `_update_nested_dict_with_flat_dict(self, params_dict: dict, flat_dict: dict) -> dict` method.
I think the problem with `_convert_dict_flat_to_nested` is that once we have two dicts, standart python dict merge overrides values with same keys. As example
`dict_1 = {"model" : {"learning_rate": 3e-4}}
dict_2 = {"model" : {"depth" : 8}}
dict_1.update(dict_2)`
will result in 
`dict_1 = {"model" : {"depth" : 8}}`
intstead of desired {"learning_rate": 3e-4, "model" : {"depth" : 8}}`
So I believe the private method should perform two operations simultaneously: convert from `"model.param" : val` to `"model" : {"param" : val}` and immediately update `params_dict`.